### ISSUE TITLE
Source Control: Select all/none repositories

### ISSUE THREAD

**Issue by adegoodyer** [microsoft/vscode#232027]
When opening vscode in a directory - all of the repos in that directory are not always added to the source control window - this tends to happen with directories that have many (10,20,30) repos.

So initially I think it would be a good idea to do this by default (if it doesn't already - needs checking for large number of repos).

More importantly - I currently have to navigate to Source Control > ... > View and Sort > Repositories > and then select a repo I want to add. Frustratingly the nested windows don't stay open or allow multiple selections! So for 30/40 repos these steps have to be repeated for each repo.

Even worse this'll reset in a couple days time so have to repeat all over again!!

If there has been user selections/deselections or suspected non vscode detection outlined above - then would be extremely useful to provide the user with an option to Source Control > ... > View and Sort > Repositories > 'select all' or 'select none' to quickly add or clear repos.!

![Image](https://github.com/user-attachments/assets/dcb4730b-1583-44a2-9f8e-30ff98ac0171)

**Comment by gjsjohnmurray** [comment]
Do you know about the SOURCE CONTROL REPOSITORIES view? You can enable this by right-clicking on the titlebar of the SOURCE CONTROL view container, then select/deselect repos in the list it displays.

**Comment by adegoodyer** [comment]
Aha, I did not! Well.. I did, but since forgotten about.

So I can now see if you click on a repo in SOURCE CONTROL REPOSITORIES then the SOURCE CONTROL window displays your selection - single, multi-click or ctrl/cmd + a for all. That last bit being the behaviour I needed.

So that's not a bad workaround - but I remember why I'd forgotten about it now - I couldn't understand the use of having both window views - for example when using SOURCE VIEW - it's essentially the same as SOURCE CONTROL REPOSITORIES but you have the additional functionality of stage control, commit authoring etc.

So.. that said - and now I understand it a bit more - would it make more sense to drop the SOURCE CONTROL REPOSITORIES view and make the SOURCE VIEW show all repos by default, with streamlined selection? That would seem a cleaner and more intuitive UX to me but would be interested to see what other people use/think?
